software/control/core.py

```diff
@@ -2320,19 +2322,20 @@ def save_image(self, image, file_ID, config, current_path):
         iio.imwrite(saving_path,image)
 
     def update_napari(self, image, config_name, i, j, k):
-        i = -1 if i is None else i
-        j = -1 if j is None else j
-        print("update napari:", i, j, k, config_name)
-
-        if USE_NAPARI_FOR_MULTIPOINT or USE_NAPARI_FOR_TILED_DISPLAY:
-            if not self.init_napari_layers:
-                print("init napari layers")
-                self.init_napari_layers = True
-                self.napari_layers_init.emit(image.shape[0],image.shape[1], image.dtype)
-            self.napari_layers_update.emit(image, i, j, k, config_name)
-        if USE_NAPARI_FOR_MOSAIC_DISPLAY and k == 0:
-            print(f"Updating mosaic layers: x={self.navigationController.x_pos_mm:.6f}, y={self.navigationController.y_pos_mm:.6f}")
-            self.napari_mosaic_update.emit(image, self.navigationController.x_pos_mm, self.navigationController.y_pos_mm, k, config_name)
+        if not self.performance_mode:
```

Comment:
This may be something that's because of my relative unfamiliarity with qt, but since we're disconnecting the napari widgets when in performance mode, can we just leave the emits here and not need this `if not self.performance_mode` check?  Since there won't be any slots listening it's hopefully effectively no-op performance-wise.

Reply:
If not, is this sufficient for performance mode and we can just not need to deal with the signal/slot connect/disconnect dance?  AKA just don't emit when in performance mode?

Reply:
youre correct, this is redundant